src/javascript/_common/base/login.js

```diff
@@ -14,9 +17,13 @@ const Login = (() => {
     const loginUrl = () => {
         const server_url = localStorage.getItem('config.server_url');
         const language   = getLanguage();
+        const signup_device      = LocalStore.get('signup_device') || (isMobile() ? 'mobile' : 'desktop');
+        const date_first_contact = moment(LocalStore.get('date_first_contact'), 'YYYY-mm-dd').valueOf();
+        const marketing_queries   = `&signup_device=${signup_device}${date_first_contact && `&date_first_contact=${date_first_contact}`}`;
+
         return ((server_url && /qa/.test(server_url)) ?
             `https://www.${server_url.split('.')[1]}.com/oauth2/authorize?app_id=${getAppId()}&l=${language}` :
```

Comment:
you need to add it to qa server url as well, as we need it for testing

Reply:
Ah, MB. right away.